source/device/receiver.cpp

```diff
@@ -70,15 +65,15 @@ std::shared_ptr<ILink> Receiver::get_link_to_destination(
 DeviceType Receiver::get_type() const { return DeviceType::RECEIVER; }
 
 Time Receiver::process() {
-    std::shared_ptr<ILink> current_inlink = m_router->next_inlink();
+    std::weak_ptr<ILink> current_inlink = m_router->next_inlink();

```

Comment:
You may use `next_inlink()` method and get `shared_ptr`:
```suggestion
    std::shared_ptr<ILink> current_inlink = next_inlink();

```